Automattic/wp-calypso
# Jetpack Pro Dash: Add ability to attach/use WC licenses

## Proposed Changes

* Creates a new route for a downloads page.
* Injects the new downloads page into the assignment flow whenever a WC license is issued.
* Provides download links and a description of what to do next.
* Modifies the `AssignmentLicenseStepProgress` component to create a new "Download products" step.
* Adds a basic util to determine whether products are WC products based on product slug or product key.
* Adds a basic util to determine a product slug based on a product key.

## To Do

* This PR was a lot more involved than expected because of the complexity of the multi-assign mechanisms. We need to test edge cases and different flows to ensure that nothing breaks.
* Need to ask for design to do a touch up on the downloads page ... I didn't do any design, only added the necessary information.

## Testing Instructions

* Use billing scheme 871 with your partner key (reach out to me if you need this switched).
* Apply this PR and `yarn start-jetpack-cloud`.
* Issue and assign a license for a single WC product. You should see the download screen and be able to download the product.
* Issue and assign licenses for two WC products. You should see the download screen and be able to download both products.
* Issue and assign licenses for any combinations of WC products and Jetpack products. You should see the appropriate download screen.
* Issue licenses for only Jetpack products. You SHOULD NOT see the download screen after assigning.
* Try to break this in any other creative ways you can think of.

## Pre-merge Checklist
